Source/WebCore/page/CaptionUserPreferencesMediaAF.cpp

```diff
@@ -196,7 +196,7 @@ void CaptionUserPreferencesMediaAF::platformSetCaptionDisplayMode(CaptionDisplay
 
 void CaptionUserPreferencesMediaAF::setCaptionDisplayMode(CaptionUserPreferences::CaptionDisplayMode mode)
 {
-    if (testingMode() || !MediaAccessibilityLibrary()) {
+    if (testingMode() || !MediaAccessibilityLibrary() || nullCaptionProfile()) {
```

Comment:
`testingMode()` is virtual. Would it work override the method in `CaptionUserPreferencesMediaAF` and call `nullCaptionProfile()` there?